dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt

```diff
@@ -32,18 +34,25 @@ internal class DataUploadRunnable(
     private val systemInfoProvider: SystemInfoProvider,
     internal val uploadSchedulerStrategy: UploadSchedulerStrategy,
     internal val maxBatchesPerJob: Int,
-    private val internalLogger: InternalLogger
+    private val internalLogger: InternalLogger,
+    private val uploadQualityListener: UploadQualityListener
 ) : UploadRunnable {
 
+    private var delayMs: Long = 0
+    private var batchConsumerAvailableAttempts: Int = 0
```

Comment:
I think we shouldn't extract these 2 as properties, this will make `DataUploadRunnable` stateful and things may become dangerous.

Let's keep them as method-local if possible?